contracts/StrategyCurveTwoPool.sol

```diff
@@ -0,0 +1,383 @@
+
+  function estimatedTotalAssets() public view override returns (uint256) {
+    return balanceOfWant().add(stakedBalance());
+  }
+
+  /* ========== MUTATIVE FUNCTIONS ========== */
+  // these should stay the same across different wants.
+
+  function adjustPosition(uint256 _debtOutstanding) internal override {
+    if (emergencyExit) {
+      return;
+    }
+    // Send all of our LP tokens to deposit to the gauge if we have any
+    uint256 _toInvest = balanceOfWant();
+    if (_toInvest > 0) {
+      gauge.deposit(_toInvest);
+    }
+  }
+
+  function liquidatePosition(uint256 _amountNeeded)
+    internal
+    override
+    returns (uint256 _liquidatedAmount, uint256 _loss)
+  {
+    uint256 _wantBal = balanceOfWant();
+    if (_amountNeeded > _wantBal) {
+      // check if we have enough free funds to cover the withdrawal
+      uint256 _stakedBal = stakedBalance();
+      if (_stakedBal > 0) {
+        gauge.withdraw(Math.min(_stakedBal, _amountNeeded.sub(_wantBal)));
+      }
+      uint256 _withdrawnBal = balanceOfWant();
+      _liquidatedAmount = Math.min(_amountNeeded, _withdrawnBal);
+      _loss = _amountNeeded.sub(_liquidatedAmount);
+    } else {
+      // we have enough balance to cover the liquidation available
+      return (_amountNeeded, 0);
+    }
+  }
+
+  function liquidateAllPositions() internal override returns (uint256) {
+    uint256 _stakedBal = stakedBalance();
+    if (_stakedBal > 0) {
+      // don't bother withdrawing zero
+      gauge.withdraw(_stakedBal);
+    }
+    return balanceOfWant();
+  }
+
+    function _claimRewards() internal {
+        gaugeFactory.mint(address(gauge));
+    }
+
+    function claimRewards() external onlyVaultManagers {
+        // Claims any pending CRV
+        //
+        // Mints claimable CRV from the factory gauge. Reward tokens are sent to `msg.sender`
+        // The method claim_rewards() from the old gauge now only applies to third-party tokens.
+        // There are no third-party tokens in this strategy.
+        _claimRewards();
+    }
+
+  function prepareMigration(address _newStrategy) internal override {
+        // Withdraw LP tokens from the gauge. The transfer to the new strategy is done
+        // by migrate() in BaseStrategy.sol. Note that this function does not claim
+        // any pending rewards. To claim and transfer pending rewards to a new
+        // strategy, please use claimAndTransferRewards()
+    uint256 _stakedBal = stakedBalance();
+    if (_stakedBal > 0) {
+      gauge.withdraw(_stakedBal);
+    }
+  }
+
+  function protectedTokens()
+    internal
+    view
+    override
+    returns (address[] memory)
+  {}
+
+  /* ========== SETTERS ========== */
+
+  // These functions are useful for setting parameters of the strategy that may need to be adjusted.
+
+  // Credit threshold is in want token, and will trigger a harvest if credit is large enough.
+  function setCreditThreshold(uint256 _creditThreshold)
+    external
+    onlyEmergencyAuthorized
```

Comment:
Any reason for `onlyEmergencyAuthorized` vs `onlyVaultManagers`? If this is just what I had in the past, then I would say the latter makes more sense. Basically `onlyVaultManagers` is the best thing to use for stuff like this imo.

Reply:
yeah, it was there. I'll update to `onlyVaultManagers`